From 78835b187a4e0eb906b9b79f5e7ca995bebc2a46 Mon Sep 17 00:00:00 2001
From: Hugues Delorme <delorme.hugues@fougsys.fr>
Date: Tue, 26 Jul 2011 16:14:22 +0200
Subject: [PATCH 1/3] [warning-fix][deleting-void*-is-undefined]

GCC reports many "deleting void* is undefined" warnings in
module TKOpenGl. This commit fixes them by using free() instead
of the delete operator generated by macro IMPLEMENT_MEMORY_OPERATORS.
This macro is defined in InterfaceGraphic_telem.hxx, the delete
operator provided is simply a wrapper around free(). So using free()
does the trick for those warnings.
---
 src/OpenGl/OpenGl_addnames.cxx        |    2 +-
 src/OpenGl/OpenGl_bintcol.cxx         |    2 +-
 src/OpenGl/OpenGl_bsurfprop.cxx       |    2 +-
 src/OpenGl/OpenGl_charexpan.cxx       |    2 +-
 src/OpenGl/OpenGl_charspace.cxx       |    2 +-
 src/OpenGl/OpenGl_degmodel.cxx        |    3 ++-
 src/OpenGl/OpenGl_edgecol.cxx         |    2 +-
 src/OpenGl/OpenGl_edgewid.cxx         |    2 +-
 src/OpenGl/OpenGl_indexpolygons.cxx   |    2 +-
 src/OpenGl/OpenGl_intcol.cxx          |    2 +-
 src/OpenGl/OpenGl_lightstate.cxx      |    2 +-
 src/OpenGl/OpenGl_localtran3.cxx      |    2 +-
 src/OpenGl/OpenGl_mrkr.cxx            |    2 +-
 src/OpenGl/OpenGl_mrkrcol.cxx         |    2 +-
 src/OpenGl/OpenGl_mrkrset.cxx         |    2 +-
 src/OpenGl/OpenGl_mrkrsize.cxx        |    2 +-
 src/OpenGl/OpenGl_polygon.cxx         |    2 +-
 src/OpenGl/OpenGl_polygonholes.cxx    |    2 +-
 src/OpenGl/OpenGl_polygonoffset.cxx   |    2 +-
 src/OpenGl/OpenGl_polyl.cxx           |    2 +-
 src/OpenGl/OpenGl_polylcol.cxx        |    2 +-
 src/OpenGl/OpenGl_polylwid.cxx        |    2 +-
 src/OpenGl/OpenGl_qstrip.cxx          |    2 +-
 src/OpenGl/OpenGl_remnames.cxx        |    2 +-
 src/OpenGl/OpenGl_surfprop.cxx        |    2 +-
 src/OpenGl/OpenGl_text.cxx            |    2 +-
 src/OpenGl/OpenGl_textcol.cxx         |    2 +-
 src/OpenGl/OpenGl_textcolsubtitle.cxx |    2 +-
 src/OpenGl/OpenGl_textheight.cxx      |    2 +-
 src/OpenGl/OpenGl_tmesh.cxx           |    2 +-
 30 files changed, 31 insertions(+), 30 deletions(-)

Index: oce/src/OpenGl/OpenGl_addnames.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_addnames.cxx
+++ oce/src/OpenGl/OpenGl_addnames.cxx
@@ -76,7 +76,7 @@
 AddNamesetDelete( TSM_ELEM_DATA data, Tint n, cmn_key *k )
 {
   if (data.pdata)
-    delete data.pdata;
+    free(data.pdata);
   return TSuccess;
 }
 
Index: oce/src/OpenGl/OpenGl_bintcol.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_bintcol.cxx
+++ oce/src/OpenGl/OpenGl_bintcol.cxx
@@ -70,7 +70,7 @@
 BackInteriorColourDelete( TSM_ELEM_DATA data, Tint n, cmn_key *k )
 {
   if (data.pdata)
-    delete data.pdata;
+    free(data.pdata);
   return TSuccess;
 }
 
Index: oce/src/OpenGl/OpenGl_bsurfprop.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_bsurfprop.cxx
+++ oce/src/OpenGl/OpenGl_bsurfprop.cxx
@@ -66,7 +66,7 @@
 BackSurfaceAreaPropertiesDelete( TSM_ELEM_DATA data, Tint n, cmn_key *k )
 {
   if (data.pdata)
-    delete data.pdata;
+    free(data.pdata);
   return TSuccess;
 }
 
Index: oce/src/OpenGl/OpenGl_charexpan.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_charexpan.cxx
+++ oce/src/OpenGl/OpenGl_charexpan.cxx
@@ -66,7 +66,7 @@
 CharacterExpansionFactorDelete( TSM_ELEM_DATA data, Tint n, cmn_key *k )
 {
   if (data.pdata)
-    delete data.pdata;
+    free(data.pdata);
   return TSuccess;
 }
 
Index: oce/src/OpenGl/OpenGl_charspace.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_charspace.cxx
+++ oce/src/OpenGl/OpenGl_charspace.cxx
@@ -66,7 +66,7 @@
 CharacterSpacingDelete( TSM_ELEM_DATA data, Tint n, cmn_key *k )
 {
   if (data.pdata)
-    delete data.pdata;
+    free(data.pdata);
   return TSuccess;
 }
 
Index: oce/src/OpenGl/OpenGl_degmodel.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_degmodel.cxx
+++ oce/src/OpenGl/OpenGl_degmodel.cxx
@@ -49,7 +49,8 @@
 static TStatus DegenerateModelDelete  ( TSM_ELEM_DATA d, Tint n, cmn_key* k ) {
 
   //cmn_freemem ( d.pdata );
-  delete d.pdata;
+  if (d.pdata)
+    free(d.pdata);
 
   return TSuccess;
 
Index: oce/src/OpenGl/OpenGl_edgecol.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_edgecol.cxx
+++ oce/src/OpenGl/OpenGl_edgecol.cxx
@@ -71,7 +71,7 @@
 EdgeColourDelete( TSM_ELEM_DATA data, Tint n, cmn_key *k )
 {
   if (data.pdata)
-    delete data.pdata;
+    free(data.pdata);
   return TSuccess;
 }
 
Index: oce/src/OpenGl/OpenGl_edgewid.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_edgewid.cxx
+++ oce/src/OpenGl/OpenGl_edgewid.cxx
@@ -69,7 +69,7 @@
 {
   //cmn_freemem( data.pdata );
   if (data.pdata)
-    delete data.pdata;
+    free(data.pdata);
   return TSuccess;
 }
 
Index: oce/src/OpenGl/OpenGl_indexpolygons.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_indexpolygons.cxx
+++ oce/src/OpenGl/OpenGl_indexpolygons.cxx
@@ -849,7 +849,7 @@
 
   //cmn_freemem( data.pdata );
   if (data.pdata)
-    delete data.pdata;
+    free(data.pdata);
   return TSuccess;
 }
 
Index: oce/src/OpenGl/OpenGl_intcol.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_intcol.cxx
+++ oce/src/OpenGl/OpenGl_intcol.cxx
@@ -74,7 +74,7 @@
 {
   //cmn_freemem( data.pdata );
   if (data.pdata)
-    delete data.pdata;
+    free(data.pdata);
   return TSuccess;
 }
 
Index: oce/src/OpenGl/OpenGl_lightstate.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_lightstate.cxx
+++ oce/src/OpenGl/OpenGl_lightstate.cxx
@@ -102,7 +102,7 @@
 static  TStatus
 LightSrcStateDelete( TSM_ELEM_DATA data, Tint n, cmn_key *k )
 {
-  delete [] data.pdata;
+  free(data.pdata);
   return TSuccess;
 }
 
Index: oce/src/OpenGl/OpenGl_localtran3.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_localtran3.cxx
+++ oce/src/OpenGl/OpenGl_localtran3.cxx
@@ -77,7 +77,7 @@
 {
   //cmn_freemem( data.pdata );
   if (data.pdata)
-    delete data.pdata;
+    free(data.pdata);
   return TSuccess;
 }
 
Index: oce/src/OpenGl/OpenGl_mrkr.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_mrkr.cxx
+++ oce/src/OpenGl/OpenGl_mrkr.cxx
@@ -166,7 +166,7 @@
 MarkerDelete( TSM_ELEM_DATA data, Tint n, cmn_key *k )
 {
   if (data.pdata)
-    delete data.pdata;
+    free(data.pdata);
   return TSuccess;
 }
 
Index: oce/src/OpenGl/OpenGl_mrkrcol.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_mrkrcol.cxx
+++ oce/src/OpenGl/OpenGl_mrkrcol.cxx
@@ -74,7 +74,7 @@
 {
   //cmn_freemem( data.pdata );
   if (data.pdata)
-    delete data.pdata;
+    free(data.pdata);
   return TSuccess;
 }
 
Index: oce/src/OpenGl/OpenGl_mrkrset.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_mrkrset.cxx
+++ oce/src/OpenGl/OpenGl_mrkrset.cxx
@@ -191,7 +191,7 @@
 MarkerSetDelete( TSM_ELEM_DATA data, Tint n, cmn_key *k )
 {
   if (data.pdata)
-    delete data.pdata;
+    free(data.pdata);
   return TSuccess;
 }
 
Index: oce/src/OpenGl/OpenGl_mrkrsize.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_mrkrsize.cxx
+++ oce/src/OpenGl/OpenGl_mrkrsize.cxx
@@ -68,7 +68,7 @@
 PolymarkerSizeDelete( TSM_ELEM_DATA data, Tint n, cmn_key *k )
 {
   if (data.pdata)
-    delete data.pdata;
+    free(data.pdata);
   return TSuccess;
 }
 
Index: oce/src/OpenGl/OpenGl_polygon.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_polygon.cxx
+++ oce/src/OpenGl/OpenGl_polygon.cxx
@@ -989,7 +989,7 @@
     delete p->display;
   }
   if (data.pdata)
-    delete data.pdata;
+    free(data.pdata);
   return TSuccess;
 }
 
Index: oce/src/OpenGl/OpenGl_polygonholes.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_polygonholes.cxx
+++ oce/src/OpenGl/OpenGl_polygonholes.cxx
@@ -650,7 +650,7 @@
 
 
   if (data.pdata)
-    delete data.pdata;
+    free(data.pdata);
   return TSuccess;
 }
 
Index: oce/src/OpenGl/OpenGl_polygonoffset.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_polygonoffset.cxx
+++ oce/src/OpenGl/OpenGl_polygonoffset.cxx
@@ -74,7 +74,7 @@
 PolygonOffsetDelete( TSM_ELEM_DATA data, Tint n, cmn_key *k )
 {
   if (data.pdata)
-    delete data.pdata;
+    free(data.pdata);
   return TSuccess;
 }
 
Index: oce/src/OpenGl/OpenGl_polyl.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_polyl.cxx
+++ oce/src/OpenGl/OpenGl_polyl.cxx
@@ -236,7 +236,7 @@
     delete[] p->vcolours;
 #endif
   if (data.pdata)
-    delete data.pdata;
+    free(data.pdata);
 
   return TSuccess;
 }
Index: oce/src/OpenGl/OpenGl_polylcol.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_polylcol.cxx
+++ oce/src/OpenGl/OpenGl_polylcol.cxx
@@ -72,7 +72,7 @@
 PolylineColourDelete( TSM_ELEM_DATA data, Tint n, cmn_key *k )
 {
   if (data.pdata)
-    delete data.pdata;
+    free(data.pdata);
   return TSuccess;
 }
 
Index: oce/src/OpenGl/OpenGl_polylwid.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_polylwid.cxx
+++ oce/src/OpenGl/OpenGl_polylwid.cxx
@@ -70,7 +70,7 @@
 PolylineWidthDelete( TSM_ELEM_DATA data, Tint n, cmn_key *k )
 {
   if (data.pdata)
-    delete data.pdata;
+    free(data.pdata);
   return TSuccess;
 }
 
Index: oce/src/OpenGl/OpenGl_qstrip.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_qstrip.cxx
+++ oce/src/OpenGl/OpenGl_qstrip.cxx
@@ -920,7 +920,7 @@
     delete[] p->vtexturecoord;
 
   if (data.pdata)
-    delete data.pdata;
+    free(data.pdata);
 
   return TSuccess;
 }
Index: oce/src/OpenGl/OpenGl_remnames.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_remnames.cxx
+++ oce/src/OpenGl/OpenGl_remnames.cxx
@@ -77,7 +77,7 @@
 RemoveNamesetDelete( TSM_ELEM_DATA data, Tint n, cmn_key *k )
 {
   if (data.pdata)
-    delete data.pdata;
+    free(data.pdata);
   return TSuccess;
 }
 
Index: oce/src/OpenGl/OpenGl_surfprop.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_surfprop.cxx
+++ oce/src/OpenGl/OpenGl_surfprop.cxx
@@ -72,7 +72,7 @@
 {
   //cmn_freemem( data.pdata );
   if (data.pdata)
-    delete data.pdata;
+    free(data.pdata);
   return TSuccess;
 }
 
Index: oce/src/OpenGl/OpenGl_text.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_text.cxx
+++ oce/src/OpenGl/OpenGl_text.cxx
@@ -371,7 +371,7 @@
 TextDelete( TSM_ELEM_DATA data, Tint n, cmn_key *k )
 {
   if (data.pdata)
-    delete data.pdata;
+    free(data.pdata);
   return TSuccess;
 }
 
Index: oce/src/OpenGl/OpenGl_textcol.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_textcol.cxx
+++ oce/src/OpenGl/OpenGl_textcol.cxx
@@ -112,7 +112,7 @@
 {
   //cmn_freemem( data.pdata );
   if (data.pdata)
-    delete data.pdata;
+    free(data.pdata);
   return TSuccess;
 }
 
Index: oce/src/OpenGl/OpenGl_textcolsubtitle.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_textcolsubtitle.cxx
+++ oce/src/OpenGl/OpenGl_textcolsubtitle.cxx
@@ -109,7 +109,7 @@
 {
   //cmn_freemem( data.pdata );  
   if (data.pdata)
-    delete data.pdata;
+    free(data.pdata);
   return TSuccess;
 }
 
Index: oce/src/OpenGl/OpenGl_textheight.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_textheight.cxx
+++ oce/src/OpenGl/OpenGl_textheight.cxx
@@ -119,7 +119,7 @@
 TextHeightDelete( TSM_ELEM_DATA data, Tint n, cmn_key *k )
 {
   if (data.pdata)
-    delete data.pdata;
+    free(data.pdata);
   return TSuccess;
 }
 
Index: oce/src/OpenGl/OpenGl_tmesh.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_tmesh.cxx
+++ oce/src/OpenGl/OpenGl_tmesh.cxx
@@ -783,7 +783,7 @@
     delete[] p->d.bDraw;
 #endif  /* G003 */
   if (data.pdata)
-    delete data.pdata;
+    free(data.pdata);
   return TSuccess;
 }
 
